test/TestProjects/MgmtExpandResourceTypes/Generated/Models/MemoryType.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace MgmtExpandResourceTypes.Models
+{
+    /// <summary> The MemoryType. </summary>
+    public enum MemoryType
+    {
+        /// <summary> -1. </summary>
+        _1 = -1,
+        /// <summary> 2. </summary>
+        Two = 2,
+        /// <summary> 4. </summary>
+        Four = 4
```

Comment:
Assigning the long value to enum int may cause out of range issue when the value is very big.

Reply:
correct. we should declare the enum as long, e.g. `public enum MemoryType : long`.